File: python-prov/src/local_part_of_id_with_space.py
from prov.model import ProvDocument

from test_case import TestCase


class LocalPartOfIdWithSpace(TestCase):

    def __init__(self):
        self.filename = "local_part_of_id_with_space"

    def create_document(self):
        document = ProvDocument()
        document.add_namespace("ex", "https://example.org/")
        document.entity("ex:a b c")
        return document

File: java-prov/src/main/java/cz/muni/fi/bthesis/LocalPartOfIdWithSpace.java
package cz.muni.fi.bthesis;

import org.openprovenance.prov.model.Entity;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.vanilla.Document;
import org.openprovenance.prov.vanilla.ProvFactory;


public class LocalPartOfIdWithSpace extends TestCase {

    public LocalPartOfIdWithSpace() {
        setFilename("local_part_of_id_with_space");
    }

    @Override
    public Document createDocument() {
        var factory = new ProvFactory();
        var document = new Document();
        var ns = new Namespace();
        ns.addKnownNamespaces();
        ns.register("ex", "https://example.org/");
        var eqn = ns.qualifiedName("ex", "a b c", factory);
        Entity entity = factory.newEntity(eqn);
        document.getStatementOrBundle().add(entity);
        document.setNamespace(ns);
        return document;
    }
}
